static/js/containers/pages/checkout/CartPage.js

```diff
@@ -0,0 +1,276 @@
+
+import type { RouterHistory } from "react-router"
+import moment from "moment"
+import { formatPrettyDateTimeAmPmTz, parseDateString } from "../../../lib/util"
+import { Button } from "reactstrap"
+
+type Props = {
+  history: RouterHistory,
+  cartItems: Array<BasketItem>,
+  isLoading: boolean
+}
+
+export class CartPage extends React.Component<Props> {
+  renderCartItemCard(cartItem: BasketItem) {
+    const purchasableObject = cartItem.product.purchasable_object
+    const course = purchasableObject.course
+
+    const title =
+      course !== undefined ? (
+        <a href="#" target="_blank" rel="noopener noreferrer">
+          {course.title}
+        </a>
+      ) : (
+        <a href="#" target="_blank" rel="noopener noreferrer">
+          {cartItem.product.description}
+        </a>
+      )
+
+    const readableId =
+      course !== undefined
+        ? purchasableObject.readable_id
+        : purchasableObject.run_tag
+
+    let startDate = ""
+    let startDateDescription = ""
+
+    if (purchasableObject.start_date) {
+      const now = moment()
+      startDate = parseDateString(purchasableObject.start_date)
+      const formattedStartDate = formatPrettyDateTimeAmPmTz(startDate)
+      startDateDescription = now.isBefore(startDate) ? (
+        <span>Starts - {formattedStartDate}</span>
+      ) : (
+        <span>
+          <strong>Active</strong> from {formattedStartDate}
+        </span>
+      )
+    }
+
+    const courseImage =
+      course !== undefined && course.page !== null ? (
+        <img src={course.page.feature_image_src} alt={course.title} />
+      ) : (
+        ""
+      )
+    const cardKey = `cartsummarycard_${cartItem.id}`
+
+    return (
+      <div
+        className="enrolled-item container card mb-4 rounded-0 flex-grow-1"
+        key={cardKey}
+      >
+        <div className="row d-flex flex-sm-columm p-md-3">
+          <div className="img-container">{courseImage}</div>
+
+          <div className="flex-grow-1 mx-3 d-sm-flex flex-column">
+            <h5 className="mt-2">{title}</h5>
+            <div className="detail">
+              {readableId}
+              <br />
+              {startDateDescription !== undefined ? startDateDescription : ""}
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  renderEmptyCartCard() {
+    return (
+      <div
+        className="enrolled-item container card mb-4 rounded-0 flex-grow-1"
+        key="cartsummarycard"
+      >
+        <div className="row d-flex flex-sm-columm p-md-3">
+          <div className="flex-grow-1 mx-3 d-sm-flex flex-column">
+            <div className="detail">Your cart is empty.</div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  renderOrderSummaryCard() {
+    let totalPrice = 0
+    const { cartItems } = this.props
+
+    if (cartItems !== null) {
+      for (let i = 0; i < cartItems.length; i++) {
+        totalPrice += Number(cartItems[i].product.price)
+      }
+    }
```

Comment:
We should move this computation to the backend - `BasketWithProductSerializer` should return the cart total so that the backend is the authoritative source of that value and we don't have this computation in both places. That way in the future when we integrate other things that effect the total (e.g. discounts), the frontend just gets that logic incorporated for free.